http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -584,6 +589,25 @@ public void testResponseStatusCode()
         assertEquals(statusCode, 543);
     }
 
+    @Test
+    public void testStreamingResponseHandler()
+            throws Exception
+    {
+        String content = IntStream.range(0, 16_000).mapToObj(ignore -> "x").collect(Collectors.joining());
```

Comment:
How long does the test take if we make this 100MB? The above seems too small to really verify that streaming large responses is working.

Reply:
Interesting result from that. 100MB allocates fast, let's keep that. But, it exposed HttpClient's executeAsync buffers the response. I've changed the code to throw an exception if `executeAsync` is used with `StreamingResponseHandler`. This seems reasonable to me. If you wanted async streaming you'd use a Virtual thread anyway.

Reply:
Update: 100MB was breaking CI (OOM). I changed to 10MB.